astrovascpy/scipy_petsc_conversions.py

```diff
@@ -365,31 +362,21 @@ def array2PETScVec(v):
 
     # slice of the global vector that belongs to this mpi rank (range: from -> to)
     nloc = iend - istart
-    # gather all ranges in rank 0, while in others the containers are None
-    Istart = MPI_COMM.gather(istart, root=0)
-    Iend = MPI_COMM.gather(iend, root=0)
+    # create a tuple of nloc in all ranks
+    Nloc = tuple(MPI_COMM.allgather(nloc))
+    # gather all ranges in a tuple, in every rank.
+    Istart = tuple(MPI_COMM.allgather(istart))
 
     # local vector
     vloc = np_zeros(nloc, PETSc.ScalarType)
 
-    if MPI_RANK == 0:
-        vloc[:nloc] = v[:nloc]
-
-    for iproc in range(1, MPI_SIZE):
-        if MPI_RANK == 0:
-            i0 = Istart[iproc]
-            i1 = Iend[iproc]
-            MPI_COMM.Send(v[i0:i1], dest=iproc, tag=77)
-        elif MPI_RANK == iproc:
-            MPI_COMM.Recv(vloc, source=0, tag=77)
-
+    # scatter the vector v on all ranks
+    MPI_COMM.Scatterv([v, Nloc, Istart, MPI.DOUBLE], vloc, root=0)  # MPIU_REAL
```

Comment:
here you can use the `from_numpy_dtype(PETSc.ScalarType)` as used here: https://github.com/BlueBrain/AstroVascPy/blob/b7e501f6ba9dfb674e12186cbc00576c69c68575/astrovascpy/scipy_petsc_conversions.py#L224

This applies because PETSc.ScalarType is numpy dtype ;)

Reply:
super!

Reply:
in my mac I have the same warning. Though in BB5 no. I would suggest you to test it on BB5 and act accordngly.

Reply:
At the end of `from_numpy_dtype` the data is duplicated by the function `Dup()`.  This causes a memory leak. 
I created a custom `_from_numpy_dtype` that fixes the problem.  It works both locally and on bb5.